python/mxnet/io.py

```diff
@@ -82,27 +82,43 @@ def get_list(shapes, types):
 class DataBatch(object):
     """A data batch.
 
+    MXNet's data iterator returns a batch of data for each `next` call.
+    This data contains `batch_size` number of examples.
+
+    If the input data consists of images then, these images should be stored in a
+    4-D matrix of shape ``(batch_size, num_channel, height, width)``.
```

Comment:
This depends on the layout. if provide_data gives DataDesc(layout='NHWC') then its (batch_size,, height, width, num_channel)